### PR TITLE
Scan address when create wallet

### LINKED ISSUE #1227: Scan ahead address when creating wallet
Create wallet with encryption by `/wallet/create` api is too slow, as it generates a encrypted wallet first, then decrypt and scan addresses, then encrypt it again. 

We should create unencrypted wallet first, then scan addresses, then encrypt it.

### PR DESCRIPTION
**Pull request by iketheadore**
Fixes #1227 

Changes:
- Scan addresses when creating wallet, only lock wallet once.

Does this change need to mentioned in CHANGELOG.md?
No

### CODE REVIEW

File: src/daemon/gateway.go [hunk 1]
```diff
@@ -538,30 +538,15 @@ func (gw *Gateway) Spend(wltID string, password []byte, coins uint64, dest ciphe
 }
 
 // CreateWallet creates wallet
-func (gw *Gateway) CreateWallet(wltName string, options wallet.Options) (*wallet.Wallet, error) {
+func (gw *Gateway) CreateWallet(wltName string, options wallet.Options, scanN uint64) (*wallet.Wallet, error) {
```
Comment (gz-c) [inline src/daemon/gateway.go]: put scanN in options?

File: src/gui/wallet.go [hunk 2]
```diff
@@ -326,12 +326,12 @@ func walletCreate(gateway Gatewayer) http.HandlerFunc {
 			}
 		}
 
-		wlt, err = gateway.ScanAheadWalletAddresses(wlt.Filename(), []byte(password), scanN-1)
-		if err != nil {
-			logger.Errorf("gateway.ScanAheadWalletAddresses failed: %v", err)
-			wh.Error500(w)
-			return
-		}
+		// wlt, err = gateway.ScanAheadWalletAddresses(wlt.Filename(), []byte(password), scanN-1)
```
Comment (gz-c) [inline src/gui/wallet.go]: remove

File: src/gui/wallet_test.go [hunk 3]
```diff
@@ -1260,8 +1236,11 @@ func TestWalletCreateHandler(t *testing.T) {
 	for _, tc := range tt {
 		t.Run(tc.name, func(t *testing.T) {
 			gateway := &GatewayerMock{}
-			gateway.On("CreateWallet", "", tc.options).Return(&tc.gatewayCreateWalletResult, tc.gatewayCreateWalletErr)
-			gateway.On("ScanAheadWalletAddresses", tc.wltName, tc.options.Password, tc.scnN-1).Return(&tc.scanWalletAddressesResult, tc.scanWalletAddressesError)
+			if tc.scnN == 0 {
+				tc.scnN = 1
+			}
+			gateway.On("CreateWallet", "", tc.options, tc.scnN).Return(&tc.gatewayCreateWalletResult, tc.gatewayCreateWalletErr)
+			// gateway.On("ScanAheadWalletAddresses", tc.wltName, tc.options.Password, tc.scnN-1).Return(&tc.scanWalletAddressesResult, tc.scanWalletAddressesError)
```
Comment (gz-c) [inline src/gui/wallet_test.go]: remove

File: src/daemon/gateway.go (hunk 1 above)
Comment (gz-c) [reply src/daemon/gateway.go]: or separate CreateWalletScanAhead method?

File: src/wallet/wallet.go [hunk 4]
```diff
@@ -178,6 +178,77 @@ func NewWallet(wltName string, opts Options) (*Wallet, error) {
 	return w, nil
 }
 
+// NewWalletWithScan create wallet and scan ahead N address
+func NewWalletWithScan(wltName string, opts Options, scanN uint64, bg BalanceGetter) (*Wallet, error) {
```
Comment (gz-c) [inline src/wallet/wallet.go]: is this almost the same as `NewWallet`?  If so, can use something like `newWallet()` internal method with an option to scan

File: src/daemon/gateway.go (hunk 1 above)
Comment (iketheadore) [reply src/daemon/gateway.go]: The CreateWallet in gateway always scan ahead addresses, so no need to have a separate method.

File: src/wallet/wallet.go [hunk 5]
```diff
@@ -178,6 +194,16 @@ func NewWallet(wltName string, opts Options) (*Wallet, error) {
 	return w, nil
 }
 
+// NewWallet creates wallet without scanning addresses
+func NewWallet(wltName string, opts Options) (*Wallet, error) {
+	return newWallet(wltName, opts, nil)
```
Comment (gz-c) [inline src/wallet/wallet.go]: error if scanN != 0

**Pull request merged**